File: WebServices_lab/src/main/java/org/example/component/column/ComplexRealColumn.java
package org.example.component.column;

import org.example.component.Column;

import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class ComplexRealColumn extends Column {

  public ComplexRealColumn(String name){
    super(name);
    this.type = ColumnType.COMPLEXREAL.name();
  }

  @Override
  public boolean validate(String data) {
    // Define a regular expression pattern for complex numbers
    String complexIntegerPattern = "^(0|[-]?0(\\.\\d+)|[-]?[1-9]\\d*(\\.\\d+)?)([-+](0(\\.\\d+)?|[1-9]\\d*(\\.\\d+)?))i$";
    Pattern pattern = Pattern.compile(complexIntegerPattern);
    Matcher matcher = pattern.matcher(data);

    // Check if the input data matches the pattern
    if (matcher.matches() || data.isEmpty()) {
      // If it matches, it's a valid complex number
      return true;
    } else {
      // If it doesn't match, it's not a valid complex number
      return false;
    }
  }
}

File: WebApplication_lab/Classes/ComplexRealColumn.cs
using System;
using System.Globalization;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Text.RegularExpressions;

namespace WebApplication.Classes
{
    public class ComplexRealColumn : Column
    {
        public ComplexRealColumn(string name) : base(name)
        {
            this.Type = ColumnType.COMPLEXREAL.ToString();
        }

        public override bool Validate(string data)
        {
            // Define a regular expression pattern for complex numbers
            string complexIntegerPattern = @"^(0|[-]?0(\.\d+)|[-]?[1-9]\d*(\.\d+)?)([-+](0(\.\d+)?|[1-9]\d*(\.\d+)?))i$";
            Regex regex = new Regex(complexIntegerPattern);
            Match match = regex.Match(data);

            // Check if the input data matches the pattern
            return match.Success || string.IsNullOrEmpty(data);
        }
    }
}
